xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceTradeService.java

```diff
@@ -372,4 +375,33 @@ public OpenOrdersParams createOpenOrdersParams() {
       throw BinanceErrorAdapter.adapt(e);
     }
   }
+
+  @Override
+  public Collection<Order> assureTradeResultsFilled(Collection<Order> orders) throws IOException {
```

Comment:
if today you did 500 trades on a given currency pair and you pass orders from yesterday here then this will fail because by default binance gives you only 500 last trades

you should set `TradeHistoryParamsTimeSpan#setStartTime` to the oldest orders timestamp on a given currency pair and you should iterate using `TradeHistoryParamsIdSpan#fromId` if the trades count you receive is > limit

Reply:
I'll look into a solution for this.

Reply:
Added a solution to this and pushed the changes.